File: modules/layout/source/types/layout_types/selector.ts
import IStyle from './styles/IStyle'
// селекторы
export class Selector {
    id: string
    name: string // значение названия селетора
    type: SelectorType // тип селетора, например класс
    styles: IStyle[] // тело селектора
    data: Selector[] // дочерние селекотры
}

// Тип селектора
export enum SelectorType {
    CLASS = 'class', // класс
    OTHER = 'other', // остальные
}

File: modules/layout/app/types/layout_types/selector.go
package layout_types

import (
    "website_layout/modules/layout/app/types/layout_types/styles"
)

// селекторы
type Selector struct {
    Id string
    Name string // значение названия селетора
    Type SelectorType // тип селетора, например класс
    Styles []styles.IStyle // тело селектора
    Data []Selector // дочерние селекотры
}

// Тип селектора
type SelectorType string

const (
    CLASS SelectorType = "class" // класс
    OTHER SelectorType = "other" // остальные
)
